http-client-core/src/main/java/io/micronaut/http/client/bind/DefaultHttpClientBinderRegistry.java

```diff
@@ -63,13 +54,16 @@
 
     private final Map<Class<? extends Annotation>, ClientArgumentRequestBinder<?>> byAnnotation = new LinkedHashMap<>();
     private final Map<Integer, ClientArgumentRequestBinder<?>> byType = new LinkedHashMap<>();
+    private final Map<Class<? extends Annotation>, AnnotatedClientRequestBinder<?>> methodByAnnotation = new LinkedHashMap<>();
 
     /**
      * @param conversionService The conversion service
-     * @param binders           The request argument binders
+     * @param argumentBinders   The request argument binders
+     * @param binders           The method binders
      */
     protected DefaultHttpClientBinderRegistry(ConversionService<?> conversionService,
-                                              List<ClientArgumentRequestBinder<?>> binders) {
+                                              List<ClientArgumentRequestBinder<?>> argumentBinders,
+                                              List<AnnotatedClientRequestBinder<?>> binders) {
```

Comment:
Think it makes sense to introduce a super interface for both `ClientArgumentRequestBinder` and `AnnotatedClientRequestBinder` maybe called `ClientRequestBinder` and inject just a single list to avoid duplicate lookups